cypher-editor-support/src/completion/AutoCompletion.js

```diff
@@ -203,7 +211,10 @@ export class AutoCompletion {
       .map(t => t.complete(types, query))
       .reduce((acc, items) => [...acc, ...items], [])
       .filter(completionItemFilter);
-
+    if (text[0] === '$') {
```

Comment:
Okey. Let's stick to this solution.

Instead of checking for `[0]` use startsWith string operation. Otherwise if string is empty you can get and error.

And in general create a new method on Autocompletion class that will filter your text.

const filteredText = AutoCompletion.filterText(text);